### DISCUSSION TITLE
/hyndsight/loocv-linear-models/

### DISCUSSION THREAD

**Discussion by robjhyndman** [robjhyndman/robjhyndman.com#177] [Announcements]
Fast computation of cross-validation in linear models 
 https://robjhyndman.com/hyndsight/loocv-linear-models/

**Comment by robjhyndman** [comment]
```header
Originally posted by Stephan Kolassa on 2014-03-17 11:59:41
````
Thanks! A couple of points:

1\) I initially looked for cv() and was a bit surprised that this
didn\'t exist\... &quot;??cv&quot; helped. It\'s a bit hard to see that CV() in
your blog post is actually two capital letters.

2\) The help page for CV() doesn\'t tell how many folds it uses (nor
does it seem possible to set this number). From your blog post, I gather
that CV() actually does LOOCV. May I suggest that this could be
mentioned in the help page to CV()?

**Reply by robjhyndman** [reply]
```header
Originally posted on 2014-03-17 21:35:50
````
Thanks Stephan.\
1. I\'ve made the font bigger.\
2. I\'ll update the help file to make this explicit. Yes, it only does
LOOCV.

**Comment by robjhyndman** [comment]
```header
Originally posted by Ricardo on 2014-03-17 23:18:02
````
Is there something similar for Lasso and ridge regression?

**Reply by robjhyndman** [reply]
```header
Originally posted on 2014-03-17 23:19:24
````
Unfortunately not.

**Reply by robjhyndman** [reply]
```header
Originally posted by pete on 2015-01-06 02:40:40
````
Thanks for putting this online. This is definitely the simplest version
of the proof I\'ve seen.

It turns out you can do this for ridge regression too. If you just
change X\'X to X\'X + NλI then the proof is basically the same.

ref: Golub, Heath, Wahba &quot;Generalized Coss-Validation as a Method for
Choosing a Good Ridge Parameter&quot;.

**Comment by robjhyndman** [comment]
```header
Originally posted by Petr Keil on 2014-03-18 04:08:57
````
Isn\'t the main purpose of AIC to approximate the take-one-out
crossvalidation statistic, so that one does not have to go through all
this?

**Reply by robjhyndman** [reply]
```header
Originally posted on 2014-03-18 04:09:56
````
Yes. And for nonlinear models, or time series models, the AIC is better
because it avoids lots of computation. But for linear regression models,
you might as well use CV directly because it is so easy to calculate.

**Reply by robjhyndman** [reply]
```header
Originally posted by Assaf on 2018-04-16 13:36:55
````
The approach in AIC and leave one out is different. The holdout in AIC
is also a conceptional idea, in addition AIC is based on a likelihood
assumption. Stone (1977) showed an asymptotically equivalent between LOO
and AIC for linear regression, however they are not the same.

**Reply by robjhyndman** [reply]
```header
Originally posted on 2018-04-16 21:54:50
````
You should read Stone (JRSSB, 1977). It is MUCH more general than that.
The AIC is asymptotically equivalent to LOOCV for ANY regression, not
just linear regression and not just Gaussian distributions. The CV
statistic has to be chosen to match the likelihood, and for Gaussian
distributions, the CV statistics is squared errors.

**Comment by robjhyndman** [comment]
```header
Originally posted by Simon Minovitsky on 2017-05-02 05:16:35
````
Hi, I looked at the source code of CV and noticed that when weights are
used in the regression the code doesn\'t use them and weights all the
residuals equally. Weighting predicted residuals squared by the weight
used in regression seems intuitive. Is mean in the code for CV
intentional?

Thank you very much.

**Reply by robjhyndman** [reply]
```header
Originally posted on 2017-05-02 05:30:54
````
Not intentional \-- I didn\'t think about weights. It would be easy
enough to adapt the code to use them.

**Reply by robjhyndman** [reply]
```header
Originally posted by Simon Minovitsky on 2017-05-02 05:40:50
````
Thank you. Yes it is very easy to change the code.I just thought maybe
you had an argument that since coefficients already calculated using
weights, weighting the residuals squared is not needed.

**Comment by robjhyndman** [comment]
```header
Originally posted by Assaf on 2018-04-16 13:19:38
````
Nice and clear description. (You have a typo in the proof, you wrote x
rather than x_i).

In addition, is this derivation is similar/related to the derivation of
GCV?

**Reply by robjhyndman** [reply]
```header
Originally posted on 2018-04-16 21:49:07
````
typo now fixed. I am unfamiliar with the derivation of GCV.

**Comment by robjhyndman** [comment]
```header
Originally posted by Simon Minovitsky on 2018-04-26 20:23:23
````
Hi,

Is it possible to derive \\betas that would minimize CV defined above?

Thank you very much,\
-Simon

**Reply by robjhyndman** [reply]
```header
Originally posted on 2018-04-29 22:38:02
````
I\'m not sure, but
[https://doi.org/10.1080/026\...](https://doi.org/10.1080/02664763.2011.646963)
is possibly relevant.

**Comment by amanjit-gill-data** [comment]
This is incredible. Thank you!
